### PR TITLE
guests-list

### PR DESCRIPTION
**Pull request by VOlekh**
**Families Page**
__________________________________________________________________________________________________________________
Trello:
https://trello.com/c/RFS4peS8/389-crate-families-list

**Families Page:**

User can push the Button I Can Help and redirect to https://wehelp.live/host.
User can push the link provided on a page to pick up the family to be hosted.
User redirected to https://wehelp.live/guests
User can't see the list of family cards with photos, id, names, and stories.
User remembers the id number and returns back to https://wehelp.live/host. and enter an id number of the family in the corresponding field
TBD: User flow to be reworked in the future. Cards' design to be changed.

**Acceptance Criteria**

- [x]  - Link leads to https://wehelp.live/guests
- [x]  - Families added to the list based on Status 'Validated' and 'Published'"
- [x]  - Following fields are present on the screen: photo, id, name, and story
- [x]  - Page made per Figma design is out of scope
- [x]  - DB clean up, loader, back button, pick up button, and pagination are out of scope

_________________________________________________________________________________________
**Demo**
https://deploy-preview-67--wehelpvip.netlify.app/guests

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,48 +1,49 @@
...
+    "start": "node .output/server/index.mjs",
+    "lint": "eslint --ext .js,.vue --ignore-path .gitignore ."
+  },
+  "lint-staged": {
+    "*.{js,vue}": "yarn lint",
+    "*.{css,vue}": "stylelint"
+  },
+  "devDependencies": {
+    "@intlify/nuxt3": "^0.1.10",
+    "@neojp/tailwindcss-line-clamp-utilities": "^1.0.1",
```
Comment (amanuela97) [inline package.json:16]: There is already a tailwind line-clamp dependency.

File: utils/airtable.js [hunk 2]
```diff
@@ -47,3 +51,27 @@ export const getCities = async (locationId) => {
...
+    const offSet = offset ? `offset=${offset}` : ''
+    const url = `https://api.airtable.com/v0/${config.BASE_ID}/Guests?&${offSet}`
+    const response = await $fetch(url, options)
+    if (response?.offset) {
+      const nextResponse = await fetchGuests(response.offset)
+      return [...response.records, ...nextResponse]
+      //   return [...response.records, ...fetchGuests(response.offset)]
+      // }
+    }
+    console.log('guests', response.records)
```
Comment (amanuela97) [inline utils/airtable.js:72]: remove console.log

File: components/guests/item.vue [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        {{ story }}
+      </div>
+    </div>
+  </div>
+</template>
+<script>
+export default {
+  name: 'GuestsItem',
+  props: {
+    id: {
```
[the remark this answers is not in the document]
Comment (VOlekh) [reply components/guests/item.vue:34]: Yes, mistyped it. Will fix

File: utils/airtable.js (hunk 2 above)
Comment (VOlekh) [reply utils/airtable.js:72]: thnks

**Review by razbakov — CHANGES_REQUESTED** [review]

**Comment by razbakov** [comment]
Please also remove empty file `components/block/family.vue`

File: tailwind.config.js [hunk 4]
```diff
@@ -143,6 +143,13 @@ module.exports = {
       },
     },
     extend: {
+      height: {
+        '128': '32rem',
+        '192': '48rem',
+        '212': '53rem',
```
Comment (razbakov) [inline tailwind.config.js:149]: in design we use 8 point grid system, that means that all numbers should be devidable by 8. we should never use odd numbers.

**Review by razbakov — CHANGES_REQUESTED** [review]

File: tailwind.config.js (hunk 4 above)
Comment (VOlekh) [reply tailwind.config.js]: I've added some more custom sizes with 8 point grid system and fixed card h

File: components/guests/list.vue [hunk 5]
```diff
@@ -0,0 +1,27 @@
+<template>
+  <div class="grid grid-cols-auto-fit gap-6 p-2">
+    <ul v-for="guest in guests" :key="guest.id" role="list">
+      <GuestsItem
+        v-if="guest.fields.Status === 'Validated' || 'Published'"
```
[the remark this answers is not in the document]
Comment (VOlekh) [reply components/guests/list.vue]: removed, with filter is really better

File: components/guests/item.vue [hunk 6]
```diff
@@ -0,0 +1,58 @@
+<template>
+  <div
+    class="col-span-1 flex flex-col h-212 p-5 bg-white-500 box-border rounded-lg border-2 border-slate-200"
+  >
+    <div class="flex flex-col  pr-3 mb-4">
```
[the remark this answers is not in the document]
Comment (VOlekh) [reply components/guests/item.vue]: ups

File: components/guests/item.vue [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+      <div v-if="photo.length !== 0" class="aspect-w-3 aspect-h-2">
```
[the remark this answers is not in the document]
Comment (VOlekh) [reply components/guests/item.vue]: got you

**Review by razbakov — CHANGES_REQUESTED** [review]

**Review by razbakov — APPROVED** [review]

**Pull request merged**